addons/project/models/account_analytic.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+    _description = 'Analytic Account'
+
+    use_tasks = fields.Boolean('Tasks', help="Check this box to manage internal activities through this project")
+    company_uom_id = fields.Many2one(related='company_id.project_time_mode_id', string="Company UOM")
+
+
+    @api.model
+    def _trigger_project_creation(self, vals):
+        '''
+        This function is used to decide if a project needs to be automatically created or not when an analytic account is created. It returns True if it needs to be so, False otherwise.
+        '''
+        return vals.get('use_tasks') and not 'project_creation_in_progress' in self.env.context
+
+    def project_create(self, vals):
```

Comment:
@atp-odoo Yes, decorator not required because this method is call from internal method.